src/Microsoft.AspNetCore.Sockets.Client/LongPollingTransport.cs

```diff
@@ -20,27 +21,36 @@ public class LongPollingTransport : ITransport
         private static readonly ProductInfoHeaderValue DefaultUserAgentHeader = ProductInfoHeaderValue.Parse(DefaultUserAgent);
 
         private readonly HttpClient _httpClient;
+        private readonly bool _ownsHttpClient;
         private readonly ILogger _logger;
         private IChannelConnection<Message> _application;
         private Task _sender;
         private Task _poller;
-        private readonly CancellationTokenSource _transportCts = new CancellationTokenSource();
+        private CancellationTokenSource _transportCts;
 
         public Task Running { get; private set; }
 
-        public LongPollingTransport(HttpClient httpClient, ILoggerFactory loggerFactory)
-        {
-            _httpClient = httpClient;
-            _logger = loggerFactory.CreateLogger<LongPollingTransport>();
-        }
+        public LongPollingTransport()
+            : this(null)
+        { }
 
-        public void Dispose()
+        public LongPollingTransport(ILoggerFactory loggerFactory)
+            : this(loggerFactory, null)
+        { }
+
+        public LongPollingTransport(ILoggerFactory loggerFactory, HttpClient httpClient)
```

Comment:
I switched the order because of Intellisense. It looked really weird when switching between these overloads. An alternative is to add an overload that only takes HttpClient but I thought it was too much. Note that loggerFactory can be null so you don't technically have to provide it.